### ISSUE TITLE
cascade_invocations should not lint awaited target

### ISSUE THREAD

**Issue by a14n** [dart-lang/linter#1317]
```dart
final list = await Future.value([]);
list.forEach(print); // LINT here shouldn't exist because it makes code ugly
```

**Comment by srawlins** [comment]
Did I fix this perhaps? I can't recreate this now.

**Comment by a14n** [comment]
Adding a test for this case is a good way to close this PR :)

**Comment by pq** [comment]
@srawlins: a test would be awesome.  Cheers @a14n!

**Comment by srawlins** [comment]
Lol I must have been lying; this had not been fixed earlier. But it is now! 😄
